pkg/render/crypto_utils.go

```diff
@@ -67,7 +67,6 @@ func CreateDexTLSSecret(dexCommonName string) *corev1.Secret {
 			Name:      DexTLSSecretName,
 			Namespace: rmeta.OperatorNamespace(),
 		},
-		Type: corev1.SecretTypeTLS,
```

Comment:
Why are we removing the secret type ?

Reply:
We are removing the type because it is not allowed to create a TLS secret without a key and it is not allowed to edit the secret type. This type was a bit of a headache case and we want to avoid that going forward with certificate management.

Reply:
You two are right. Things evolved while working on this ticket. While the SecretType is a nuisance, I will revert this change.